src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -0,0 +1,1876 @@
+//------------------------------------------------------------------------------
+/*
```

Comment:
@ximinez: this commit basically split `LedgerConsensus` into two classes: `LedgerConsensus` which is a public interface (for now, anyways) and its implementation. I then split the implementation into a .h and a .cpp and de-inlined.

Most everything should be identical; in a few places (some of which @scottschurr identified) we now use `ConsensusImp` instead of calling back to `NetworkOPs` but that's the idea: to isolate consensus state and access it via `Consensus` (for others) and `ConsensusImp` (for us)